src/python/pants/backend/python/packaging/pyoxidizer/config.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from string import Template
+from textwrap import indent
+from typing import Optional
+
+DEFAULT_TEMPLATE = """
+def make_exe():
+    dist = default_python_distribution()
+    policy = dist.make_python_packaging_policy()
+
+    # Note: Adding this for pydanic and libs that have the "unable to load from memory" error
+    # https://github.com/indygreg/PyOxidizer/issues/438
+    policy.resources_location_fallback = "filesystem-relative:lib"
+
+    python_config = dist.make_python_interpreter_config()
+    $ENTRY_POINT
+
+    exe = dist.to_python_executable(
+        name="$NAME",
+        packaging_policy=policy,
+        config=python_config,
+    )
+
+    # pip_download requires that wheels are available for each dep
+    # exe.add_python_resources(exe.pip_download($WHEELS))
+    exe.add_python_resources(exe.pip_install($WHEELS))
+    $UNCLASSIFIED_RESOURCE_INSTALLATION
+
+    return exe
+
+def make_embedded_resources(exe):
+    return exe.to_embedded_resources()
+
+def make_install(exe):
+    # Create an object that represents our installed application file layout.
+    files = FileManifest()
+    # Add the generated executable to our install layout in the root directory.
+    files.add_python_resource(".", exe)
+    return files
+
+register_target("exe", make_exe)
+register_target("resources", make_embedded_resources, depends=["exe"], default_build_script=True)
+register_target("install", make_install, depends=["exe"], default=True)
+resolve_targets()
+"""
+
+UNCLASSIFIED_RESOURCES_TEMPLATE = """
+for resource in exe.pip_install($UNCLASSIFIED_RESOURCES):
+    resource.add_location = "filesystem-relative:lib"
+    exe.add_python_resource(resource)
+"""
+
+
+class PyOxidizerConfig:
```

Comment:
We try to use `dataclasses` for cases like this, optionally with `frozen_after_init` to allow `__init__` to be defined: https://github.com/pantsbuild/pants/blob/e77191184f66e866a7282f21f40f6ad652566d62/src/python/pants/backend/project_info/dependees.py#L43-L55

Reply:
I ended up using `frozen` and just creating a property to correctly output the module - overriding it in the init felt sloppy to me.